torch/csrc/utils/mode_utils.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+#include <c10/core/ModePyObjTrampoline.h>
+#include <torch/csrc/Exceptions.h>
+#include <torch/csrc/autograd/python_variable.h>
+#include <torch/csrc/utils/python_strings.h>
+
+namespace torch {
+struct ConcreteModePyObjTrampoline : virtual c10::ModePyObjTrampoline {
```

Comment:
This should be the only sort of `ModePyObjTrampoline` to ever exist but it lets us just define `ModePyObjTrampoline` in the c10 namespace and then this in the torch namespace, where we're able to get the interpreter